### PR TITLE
ci: sync release to pg branch

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description
This CI job is responsible for syncing changes from the release branch to the pg branch. This happens for every commit merged to release branch, similar to `Sync Community Workflow`

Fixes #`Issue Number`  

## Automation

/ok-to-test tags="@tag.Sanity"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10571814611>
> Commit: 9bc44fc0cc762d52854da3b4e2f8c3e995909ccd
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10571814611&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Sanity`
> Spec:
> <hr>Tue, 27 Aug 2024 05:00:52 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced an automated workflow for merging changes from the release branch to the pg branch, enhancing integration and development efficiency.
	- Added notifications for merge conflicts to improve communication and resolution tracking.

### CODE REVIEW

**Review by abhvsn — CHANGES_REQUESTED** [review]
Changes looks good can we start notifying on postgres channel instead of tech channel to start with? This will reduce the noise for general audience.

**Comment by AnaghHegde** [comment]
@abhvsn I have changed the channel id to point to pg channel for the same reason you have mentioned in the comment.

**Comment by abhvsn** [comment]
> @abhvsn I have changed the channel id to point to pg channel for the same reason you have mentioned in the comment. Here is commit hash for reference - [f8ab124](https://github.com/appsmithorg/appsmith/commit/f8ab124282d91a2d4b95ea9edd955a7146261377)

Can we update the the comment in that case?

File: .github/workflows/sync-release-to-pg.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          for commit in $CONFLICTED_COMMITS; do
+            commit_url="$REPOSITORY_URL/commit/$commit"
+            message+="$commit_url\n"
+          done
+
+          # Send the message to Slack
+          # This unwieldy horror of a sed command, converts standard Markdown links to Slack's unwieldy link syntax.
+          slack_message="$(echo "$message" | sed -E 's/\[([^]]+)\]\(([^)]+)\)/<\2|\1>/g')"
+
+          # This is the ChannelId of the tech channel.
```
Comment (abhvsn) [inline .github/workflows/sync-release-to-pg.yml:73]: Can we update the comment to mention currently we are notifying on postgres channel instead.

**Review by abhvsn — APPROVED** [review]

**Pull request merged**